dockstore-common/src/test/java/io/dockstore/common/yaml/DockstoreYamlTest.java

```diff
@@ -198,4 +202,81 @@ public void testGalaxySubclass() throws DockstoreYamlHelper.DockstoreYamlExcepti
         assertEquals(4, workflows.stream().filter(w -> w.getSubclass().equalsIgnoreCase("gxformat2")).count());
     }
 
+    @Test
+    public void testGitReferenceFilter() {
+        assertTrue(DockstoreYamlHelper.filterGitReference(Path.of("anything"), List.of()));
+
+        // Generic name filter
+        // Glob
+        assertTrue(DockstoreYamlHelper.filterGitReference(Path.of("refs/heads/develop"), List.of("develop")));
+        assertTrue(DockstoreYamlHelper.filterGitReference(Path.of("refs/tags/develop"), List.of("develop")));
+        assertTrue(!DockstoreYamlHelper.filterGitReference(Path.of("refs/heads/foo/develop"), List.of("develop")));
+        assertTrue(!DockstoreYamlHelper.filterGitReference(Path.of("refs/tags/foo/develop"), List.of("develop")));
+        // Regex
+        assertTrue(DockstoreYamlHelper.filterGitReference(Path.of("refs/heads/develop"), List.of("/develop/")));
+        assertTrue(DockstoreYamlHelper.filterGitReference(Path.of("refs/tags/develop"), List.of("/develop/")));
+        assertTrue(!DockstoreYamlHelper.filterGitReference(Path.of("refs/heads/foo/develop"), List.of("/develop/")));
```

Comment:
Contradicting what I said in Slack, I think this should pass. The string 'develop' is in the path.

Reply:
Continuing discussion on #dockstore so we can align on desired behaviour